subsys/bootloader/cmake/sign.cmake

```diff
@@ -196,3 +196,24 @@ foreach (slot ${slots})
     )
 
 endforeach()
+
+if (CONFIG_BUILD_S1_VARIANT)
+
+  include(${NRF_DIR}/cmake/fw_zip.cmake)
+
+  # Generate zip file with both update candidates
+  set(s0_bin_path ${PROJECT_BINARY_DIR}/signed_by_b0_s0_image.bin)
+  get_filename_component(s0_name ${s0_bin_path} NAME)
+  set(s1_bin_path ${PROJECT_BINARY_DIR}/signed_by_b0_s1_image.bin)
+  get_filename_component(s1_name ${s1_bin_path} NAME)
+
+  generate_dfu_zip(
+    ${PROJECT_BINARY_DIR}/dfu_b0.zip # Zip out
+    "${s0_bin_path};${s1_bin_path}"
+    "s0_image_signed_kernel_hex_target;s1_image_signed_kernel_hex_target" # dependencies
```

Comment:
Might be out of scope, but you can use https://cmake.org/cmake/help/latest/command/cmake_parse_arguments.html
to avoid quoting the lists.

Reply:
It would also make the argument list more readable :).

Reply:
We are not inside a function at this point, and AFAICS this is intended for use inside functions and macros.

`This command is for use in macros or functions.`

Reply:
Yep, sorry, I meant use it inside `generate_dfu_zip()` to make it more idiomatically Cmake, something like:

```
  generate_dfu_zip(
    OUTPUT
    ${PROJECT_BINARY_DIR}/dfu_application.zip
    INPUTS
    ${s0_bin_path}
    ${s1_bin_path}
    DEPENDENCIES
    s0_image_signed_kernel_hex_target;s1_image_signed_kernel_hex_target
    ARGUMENTS
    "type=application"
    "${s0_name}load_address=$<TARGET_PROPERTY:partition_manager,PM_S0_ADDRESS>"
    "${s1_name}load_address=$<TARGET_PROPERTY:partition_manager,PM_S1_ADDRESS>"
    "version_B0=${CONFIG_FW_INFO_FIRMWARE_VERSION}"
    )
```

Reply:
Agree, to help the reader in this case, this should use `cmake_parse_arguments`

Reply:
Added use of cmake_parse_arguments`